governance/program/src/addins/voter_weight.rs

```diff
@@ -23,7 +23,24 @@ pub enum VoterWeightAccountType {
     VoterWeightRecord,
 }
 
-/// VoterWeight Record account
+/// The governance action VoterWeight is evaluated for
+#[derive(Clone, Debug, PartialEq, BorshDeserialize, BorshSerialize, BorshSchema)]
+pub enum VoterWeightAction {
```

Comment:
nit: since all of these actions concern a target, you can make a helper struct that has this enum + pubkey, or you can have the value contained in the enum, ie:
```
enum VoterWeightAction {
    CastVote(Pubkey),
...
}
```

Reply:
The `addin` can specify only the target without any action, or just the action without any target.
For example:
-  It can define voter's weight for anything pertaining to a `Realm` without explicit action.
- It can specify voter's weight for `CastVote` for any proposal